tests/gha-run-k8s-common.sh

```diff
@@ -264,3 +264,29 @@ function deploy_k8s() {
 
 	echo "::endgroup::"
 }
+
+function install_snapshotter() {
+    if [ "${SNAPSHOTTER}" == "nydus" ]; then
+        echo "Skipping install for ${SNAPSHOTTER} snapshotter"
+        return 0
+    fi
```

Comment:
Yes.

Reply:
So, @liudalibj has a valid point.

Depending on where we're testing, we simply do not have a simple way to:
* Add the nydus snapshotter directly to the node
* Easily edit the containerd config directly in the node (and it's because of whether the node is using rke2 / k3s / k0s / vanilla k8s ....

That's the reason I think the snapshotter daemonset should come from nydus, and should label the node correctly, and the kata-deploy would do the right thing depending on what your node was labeled.

Reply:
I've created https://github.com/containerd/nydus-snapshotter/issues/565 and I'll poke nydus maintainers to take a look at it, and see if we can leverage this approach.

Reply:
I'll take this.

Reply:
I was recently testing/reviewing this and noticed @fidencio's comments. Quick question: Does this mean that this PR could be closed, shifting focus towards deploying at the containerd level, or does this PR still hold relevance?

Reply:
Hi @beraldoleal , this PR is still relevant. It installs nydus-snapshotter using the daemonset introduced in https://github.com/containerd/nydus-snapshotter/tree/main/misc/snapshotter